src/main/java/com/projetogerenciamentocurso/gerenciamentocurso/service/ProfessorService.java

```diff
@@ -0,0 +1,93 @@
+package com.projetogerenciamentocurso.gerenciamentocurso.service;
+
+import javax.transaction.Transactional;
+
+import org.springframework.stereotype.Service;
+
+import com.projetogerenciamentocurso.gerenciamentocurso.GerenciamentoCursoApplication;
+import com.projetogerenciamentocurso.gerenciamentocurso.dto.ProfessorDto;
+import com.projetogerenciamentocurso.gerenciamentocurso.dtoresposta.ProfessorDtoResposta;
+import com.projetogerenciamentocurso.gerenciamentocurso.mensageria.Publisher;
+import com.projetogerenciamentocurso.gerenciamentocurso.models.Professor;
+import com.projetogerenciamentocurso.gerenciamentocurso.repository.ProfessorRepository;
+
+import lombok.AllArgsConstructor;
+
+@Service
+@AllArgsConstructor
+@Transactional
+public class ProfessorService {
+
+	
+	private final Publisher publisher;
+
+	
+	private final ProfessorRepository professorRepository;
+	
+	public ProfessorDtoResposta criarProfessor(ProfessorDto professor) {
+		
+		Professor professorModel = criarModelProfessor(professor);
+		
+		professorRepository.save(professorModel);
+		
+		publisher.send(GerenciamentoCursoApplication.EXCHANGE_NAME,
+				GerenciamentoCursoApplication.ROUTING_PROFESSOR_ATUALIZAR, professor);
+		
+		return criarProfessorDtoResposta(professor);
+		 
+	}
+
+	public ProfessorDtoResposta atuzalizarProfessor(ProfessorDto professor) {
+
+		Professor atualizaProfessor = professorRepository.getOne(professor.getIdPessoa());
+		if(atualizaProfessor!=null) {
+			
+			atualizaProfessor.setCpf(professor.getCpf());
+			atualizaProfessor.setEmail(professor.getEmail());
+			atualizaProfessor.setNome(professor.getNome());
+			atualizaProfessor.setTitulacao(professor.getTitulacao());
+			professorRepository.save(atualizaProfessor);
+			
+			publisher.send(GerenciamentoCursoApplication.EXCHANGE_NAME,
+					GerenciamentoCursoApplication.ROUTING_PROFESSOR_CRIAR, professor);
+		}
+		
+		return criarProfessorDtoResposta(professor);
+	}
+
+	public ProfessorDtoResposta deletarProfessor(ProfessorDto professor) {
+
+		Professor professorModel = criarModelProfessor(professor);

```

Comment:
não precisa criar o model, faça a busca pelo id do professor que veio como parâmetro e verifique se não for nulo, caso for lance exception

Reply:
feito